core/trino-spi/src/main/java/io/trino/spi/block/EncoderUtil.java

```diff
@@ -68,16 +72,53 @@ public static void encodeNullsAsBits(SliceOutput sliceOutput, @Nullable boolean[
         sliceOutput.writeBytes(packedIsNull, 0, currentByte);
     }
 
+    /**
+     * Implementation of {@link EncoderUtil#encodeNullsAsBitsScalar(SliceOutput, boolean[], int, int)} using the vector API
+     */
+    @SuppressWarnings({"NarrowingCompoundAssignment", "ImplicitNumericConversion"})
+    public static void encodeNullsAsBitsVectorized(SliceOutput sliceOutput, @Nullable boolean[] isNull, int offset, int length)
+    {
+        sliceOutput.writeBoolean(isNull != null);
+        if (isNull == null) {
+            return;
+        }
+        // inlined from Objects.checkFromIndexSize
+        if ((length | offset) < 0 || length > isNull.length - offset) {
+            throw new IndexOutOfBoundsException("Invalid offset: %s, length: %s for size: %s".formatted(offset, length, isNull.length));
+        }
+        byte[] packedIsNull = new byte[(length + 7) / 8];
+        int currentByte = 0;
+        int position = 0;
+        for (; position < ByteVector.SPECIES_64.loopBound(length); position += ByteVector.SPECIES_64.length(), currentByte++) {
```

Comment:
Can we write these for loops without initialiser as while loops ?
I think that would be a bit easier to read